### DISCUSSION TITLE
phpiapm first_free

### DISCUSSION THREAD

**Discussion by mcande27** [codeaffen/phpipam-ansible-modules#89] [Q&A]
Hello,
Thanks very much for this collection !

Is it possible to use the first_free function (like the API) to reserve an IP without write it in the playbook?
If not, will this be implemented in the future?

Regards.
Thanks.
Matthieu

**Comment by cmeissner** [comment]
Hi @mcande27,

Thank you for bringing this topic to our attention. If you have a look into codeaffen/phpipam-ansible-modules#34 and codeaffen/phpipam-ansible-modules#53 you can see that there are aspirations to have such features available. 
Unfortunately the development is done solely by myself and I found not enough time to implement such new features. You can file a feature request if you want to guarantee that this idea will not be lost. 
From my point of view a refactoring of the API layer is needed to remove the dependency of implementing this layer again and again. Currently the API logic is in the [module_utils.py](https://github.com/codeaffen/phpipam-ansible-modules/blob/develop/plugins/module_utils/phpipam_helper.py) which can only be used from within modules not from other components like lookup or inventory plugins.
